### PR TITLE
fix(ls): Actually respect triage ignored findings in the language server

### PR DESCRIPTION
**Pull request by ajbt200128**
It looks like we were hitting an endpoint w/out the semgrep user agent header. To fix this, we now always make requests w/ the semgrep user agent, like in pysemgrep. This adds another param to the http_helpers functor, but is imo ergonomic enough I'm ok with it (sorry pad).

## Test plan
run in vscode and verify error message no longer appears

### CODE REVIEW

File: libs/networking/http_helpers/http_helpers.ml [hunk 1]
```diff
@@ -51,8 +51,16 @@ let set_client_ref v = if not !in_mock_context then client_ref := Some v
...
   val run : 'a Lwt.t -> 'a
+end) (Version : sig
+  val version : string
 end) =
 struct
+  (* The agent is needed by many endpoints in our backend guarded by
+   * @require_supported_cli_version()
+   * alt: use Metrics_.string_of_user_agent()
+   *)
+  let user_agent_header = ("User-Agent", Fmt.str "Semgrep/%s" Version.version)
```
Comment (aryx) [inline libs/networking/http_helpers/http_helpers.ml]: the libs/ directory is supposed to contain code that is independent of semgrep, that could be used for other stuff,
so better to do this thing elsewhere.
Maybe pass the all user_agent string, not just version, so then it's independent of semgrep.

But honestly I think it's better to add the header at the get_async time, so it gives more flexibility to the caller.
Also what happens if there are 2 "User-agent" in the list of headers? which one take precedence?

**Review by aryx — CHANGES_REQUESTED** [review]
Does not feel right.
If you want to factorize things, make an helper function in Semgrep_app.ml

File: libs/networking/http_helpers/http_helpers.ml (hunk 1 above)
Comment (ajbt200128) [reply libs/networking/http_helpers/http_helpers.ml]: that's fair, I'll just switch it back to adding headers at get_async time

Comment (kopecs) [reply libs/networking/http_helpers/http_helpers.ml]: > Also what happens if there are 2 "User-agent" in the list of headers? which one take precedence?

Both would be sent.

**Review by aryx — APPROVED** [review]

**Pull request merged**